### ISSUE TITLE
Full-text search does not appear to be respecting a space (as documentation states)

### ISSUE THREAD

**Issue by isleshocky77** [bitwarden/clients#2185]
### Steps To Reproduce

1. Go to Bitwarden Add-on
2. Click on  Search
3. Type `>name:American Airlines`

### Expected Result

results with **ONLY** `name = 'American Airlines'` only.

### Actual Result

results with `name = 'American Airlines'` **AND** `name = 'Jetblue Airlines'` **AND**  `name = 'American Express'`

### Screenshots or Videos

![Screenshot from 2021-11-17 11-55-18](https://user-images.githubusercontent.com/361586/142246331-72428272-aa85-4ba5-900e-0b883c5220a5.png)

### Additional Context

According to [search documentation](https://bitwarden.com/help/article/searching-vault/#full-text-search) it looks like the space should be respected.

I've also attempted work-arounds with the following queries which have either no results or similar results:

* `>name:'American Airlines'`
* `>name:"American Airlines"`
* `>name:American+Airlines`
* `>name:American%20Airlines`

**NOTE** This is the same incorrect result as happening in https://vault.bitwarden.com

### Operating System

Linux

### Operating System Version

Ubuntu 20.04

### Web Browser

Chrome

### Browser Version

94.0.4606.71 (Official Build) (64-bit)

### Build Version

1.54.0

**Comment by isleshocky77** [comment]
This may actually belong in the [jslib repo](https://github.com/bitwarden/jslib) otherwise, it probably needs to be duplicated to the [web repo](https://github.com/bitwarden/web) as well.

**Comment by isleshocky77** [comment]
This is still an issue.

**Comment by dwbit** [comment]
Thanks for letting us know. I've removed the label.

**Comment by miguelarios** [comment]
Any way around this? I tried quotes, but that just didn't work

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by isleshocky77** [comment]
four years later... and still an issue. Not stale.

**Comment by working-name** [comment]
5 years later, still an issue. (not to mention search itself is pure 💩 if you don't tell it to look in the name: field for example. In fact, I have no clue how the results are obtained, I get back my whole vault with a simple search 😆 )

**Comment by bwbug** [comment]
I'm not sure how this thread has remained open for almost 5 years.  Bitwarden's search is based on [Lunr](https://lunrjs.com/), and has _never_ supported searching by phrase.   The search index consists of [tokens](https://lunrjs.com/guides/core_concepts.html#text-processing) created by treating space and hyphen characters as delimiters. And when searching for two (or more) terms at once, they are always [combined using OR](https://lunrjs.com/guides/searching.html), although vault items that contain _both_ terms are boosted in the search results (using [Okapi BM25 scoring](https://lunrjs.com/guides/searching.html#scoring)), similar to the way that search engines (like Google) work.

@isleshocky77 To find _only_ items that contain both "American" and "Airlines" in the item name, you would need to perform an advanced search using the following syntax:

```
>+name:american +name:airlines
```
&nbsp;

> In fact, I have no clue how the results are obtained

@working-name If you have questions about this, and don't find the answers you need in the [documentation](https://bitwarden.com/help/searching-vault/), the best place to ask is the [Bitwarden Community Forum](https://community.bitwarden.com/c/support/pm-ask-the-community/62).

**Comment by working-name** [comment]
> advanced search using the following syntax:
> 
> ```
> >+name:american +name:airlines
> ```

That is good. It could potentially be shorter and more user friendly to just take `"american airlines"` and translate the search to the advanced syntax internally. It's less fun for me to find `>` and then `+` on a mobile keyboard for example.

**Comment by bwbug** [comment]
Unfortunately, I don't think advanced search syntax is available for the mobile apps.

**Comment by isleshocky77** [comment]
> I'm not sure how this thread has remained open for almost 5 years. Bitwarden's search is based on [Lunr](https://lunrjs.com/), and has _never_ supported searching by phrase. The search index consists of [tokens](https://lunrjs.com/guides/core_concepts.html#text-processing) created by treating space and hyphen characters as delimiters. And when searching for two (or more) terms at once, they are always [combined using OR](https://lunrjs.com/guides/searching.html), although vault items that contain _both_ terms are boosted in the search results (using [Okapi BM25 scoring](https://lunrjs.com/guides/searching.html#scoring)), similar to the way that search engines (like Google) work.

@bwbug This is awesome! I'm always happy to learn something new - and I didn't know at the time - or even now - a workaround for my issue described in this issue existed. However, your solution definitely works and is completely workable.  I either didn't read the docs on the search infrastructure well enough at the time - or it didn't work (probably the former).

None the less, thank you.

Quick Edit: Before hitting submit on this comment I re-read my initial comment and noticed I specifically call out the docs. Well, five years later and the docs are still wrong.

<img width="1133" height="318" alt="Image" src="https://github.com/user-attachments/assets/03e9bd93-1f1c-40eb-a548-77e14d5a885f" />

Source: https://bitwarden.com/help/searching-vault/#full-text-search

**Comment by bwbug** [comment]
You are absolutely correct about the documentation being in error!  Unfortunately, I don't work for Bitwarden, so I have no control over their process for fixing documentation errors.  However, there is a link at the bottom of each Help Center article for [suggesting changes](https://bitwarden.com/help/searching-vault/#help-page-suggestions); I plan to submit a note there.

**Comment by bwbug** [comment]
I just submitted the following suggestion:

>The 2nd and 3rd examples under "Searching specific fields" (https://bitwarden.com/help/searching-vault/#searching-specific-fields) are wrong, as explained in the following Github comment:
https://github.com/bitwarden/clients/issues/2185#issuecomment-4238026962
>
>Lunr does not permit searching for exact phrases.  Therefore, it is not possible to search for items with the terms "Turbo Tax" or "Security Question" in the item name or custom field, or elsewhere.  For example, the closest we can get to executing a search for items with "Turbo Tax" in the name would be an advanced search expression with the following syntax:
>`>+name:turbo +name:tax`
>However, even this search expression would return a vault item with the name "Tax-Free Turbo-Prop Airplane".
>
>The "Security Question" example is even more problematic.  Not only does the search treat multi-word phrases as individual search terms, but the `>fields:` prefix searches the custom field *values* as well as their names.  The search expression in the 3rd example (an attempt to search for "Security Question" among the custom field names) will return vault items do not contain *any* custom fields, as long as the word "Question" (even by itself) appears somewhere else in the vault item (e.g., in the item name, notes, attachment file name, etc.), and it will also return a vault item that has a custom field with the field label (field name) "Job Title" and a corresponding field value of "Security Guard".
>
>Please fix these errors in the documentation.
